src/modules/core/components/TokenEditDialog/TokenEditDialog.tsx

```diff
@@ -133,13 +135,61 @@ const TokenEditDialog = ({
           <Heading appearance={{ size: 'normal' }} text={MSG.noTokensText} />
         )}
       </DialogSection>
-      <DialogSection appearance={{ align: 'right' }}>
-        <Button
-          appearance={{ theme: 'primary', size: 'large' }}
-          text={MSG.buttonDone}
-          onClick={close}
-        />
-      </DialogSection>
+      <Form
+        initialValues={{
+          tokenAddress: '',
+          description: '',
+        }}
+        onSubmit={handleSubmit}
+        validationSchema={validationSchema}
+      >
+        {({
+          isSubmitting,
+          isValid,
+          dirty,
+          values,
+        }: FormikProps<FormValues>) => (
+          <>
+            <DialogSection>
+              <Paragraph className={styles.description}>
+                <FormattedMessage {...MSG.notListedToken} />
+              </Paragraph>
+              <TokenSelector
+                tokenAddress={values.tokenAddress}
+                onTokenSelect={(token: OneToken) => handleTokenSelect(token)}
+                tokenData={tokenData}
+                label={MSG.fieldLabel}
+                appearance={{ colorSchema: 'grey' }}
+              />
+              <div className={styles.textarea}>
+                <Textarea
+                  appearance={{
+                    colorSchema: 'grey',
+                    resizable: 'vertical',
+                  }}
+                  label={MSG.textareaLabel}
+                  name="description"
+                  maxLength={4000}
+                />
+              </div>
+            </DialogSection>
+            <DialogSection appearance={{ align: 'right' }}>
+              <Button
+                appearance={{ theme: 'secondary', size: 'large' }}
+                text={MSG.buttonCancel}
+                onClick={close}
+              />
+              <Button
+                appearance={{ theme: 'primary', size: 'large' }}
+                text={MSG.buttonConfirm}
```

Comment:
We already have global id message descriptors for common buttons: https://github.com/JoinColony/colonyDapp/blob/4e02b406ae13b6ba693914fb2304d1bd83872d11/src/i18n/en.json#L33

So in this case it means you don't have to use dedicated message descriptors for the Confirm and Cancel buttons.

```suggestion
                text={{ id: 'button.confirm' }}
```

Please make these changes to the buttons and remove the message descriptors.